### PR TITLE
enhance:[Cherry-Pick] Refine varchar length exceeds max length error message (#31521)

### PR DESCRIPTION
**Pull request by aoiasd**
pr: https://github.com/milvus-io/milvus/pull/31521

### CODE REVIEW

File: internal/proxy/validate_util.go [hunk 1]
```diff
@@ -518,24 +523,24 @@ func (v *validateUtil) checkArrayFieldData(field *schemapb.FieldData, fieldSchem
...
+				return merr.WrapErrParameterInvalidMsg("the length (%d) of %dth %s %s[%d] exceeds max length (%d)",
+					len(row.GetStringData().GetData()[i]), rowCnt, fieldSchema.GetDataType().String(), fieldSchema.GetName(), i, maxLength)
 			}
 		}
 	}
 	return v.checkArrayElement(data, fieldSchema)
 }
 
-func verifyLengthPerRow[E interface{ ~string | ~[]byte }](strArr []E, maxLength int64) error {
+func verifyLengthPerRow[E interface{ ~string | ~[]byte }](strArr []E, maxLength int64) (int, bool) {
```
Comment (tedxu) [inline internal/proxy/validate_util.go:536]: Any particular reason to change the method signature here?

**Review by tedxu — COMMENTED** [review]
/lgtm

File: internal/proxy/validate_util.go (hunk 1 above)
Comment (congqixia) [reply internal/proxy/validate_util.go:536]: IMHO, to return the error index

**Pull request merged**